website_event_crm/models/crm_lead.py

```diff
@@ -65,6 +48,30 @@ def _cron_auto_invite_website_event_type(self):
             except Exception:
                 _logger.exception("Failure trying to invite to website event type.")
 
+    @api.model
+    def _cron_auto_invite_website_event_type(self):
+        """Invite automatically leads/opportunities to website event categories."""
+        # Find opportunities
+        leads = self.search(
+            [
+                ("email_from", "!=", False),
+                ("event_type_id", "!=", False),
+                ("stage_id.auto_invite_website_event_type", "=", True),
+            ]
+        )
+        for company in [x.company_id for x in leads]:
+            company_leads = leads.filtered(lambda l: l.company_id == company)
+            available_event_types = company_leads.mapped("event_type_id")
+            published_events = (
+                self.env["event.event"]
+                .search(available_event_types._published_events_domain())
+                .filtered(lambda e: e.company_id == company or not e.company_id)
+            )
+            published_event_types = published_events.mapped("event_type_id")
+            if not published_event_types:
+                return
```

Comment:
You probably need to change this too:

```suggestion
                continue
```

Reply:
Sure! :sweat_smile: